### DISCUSSION TITLE
Add support for Web Test Runner in @nx/angular:library

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [nrwl/nx#27700] [Feature Requests]
Currently, the only supported `unitTestRunner` for  [@nx/angular:library}(https://nx.dev/nx-api/angular/generators/library) is `jest`. I understand `karma` has been deprecated and is no longer available in NX.

Since Angular announced its transition to [Web Test Runner](https://modern-web.dev/docs/test-runner/overview/) (see [this announcement from the Angular Blog](https://blog.angular.dev/moving-angular-cli-to-jest-and-web-test-runner-ef85ef69ceca)) it would be nice to add it to NX as well.

This would be particularly useful since at the moment the combo Angular+Jest has a serious performance issue ( see https://github.com/thymikee/jest-preset-angular/issues/2161 ) and unit tests run quite slow.

**Comment by gperdomor** [comment]
Not only for angular libraries, I think should be supported by `@nx/js:library` too...

**Reply by RobbyRabbitman** [reply]
Hey there, I have created a plugin that infers Web Test Runner projects - check it out :) https://github.com/RobbyRabbitman/nx-plus/blob/main/libs/web-test-runner/README.md
